plugins/module_utils/foreman_helper.py

```diff
@@ -725,16 +725,15 @@ def _update_entity(self, resource, desired_entity, current_entity, params, forem
             old_value = current_entity.get(key)
             # String comparison needs extra care in face of unicode
             if foreman_type == 'str':
-                old_value = to_native(current_entity.get(key))
-                new_value = to_native(value)
-            elif foreman_type == 'list':
-                if value and isinstance(value[0], dict):
-                    if 'name' in value[0]:
-                        sort_key = 'name'
-                    else:
-                        sort_key = list(value[0].keys())[0]
-                    new_value = sorted(value, key=operator.itemgetter(sort_key))
-                    old_value = sorted(current_entity.get(key), key=operator.itemgetter(sort_key))
+                old_value = to_native(old_value)
+                new_value = to_native(new_value)
+            elif foreman_type == 'list' and foreman_spec[key].get('elements') == 'dict' and value:
+                if 'name' in value[0]:
```

Comment:
At this point, the list of repositories in the `content_view` module has been flattened into a list of `repository_ids` already.
But also i am wondering, why the `foreman_type` is not `entity_list`.

Reply:
because it's ansible type at this point, mhhh, maybe I need to revisit my logic once more

Reply:
We should stop calling the variable `foreman_spec` in that case. It creates wrong expectations.
Is it `ansible_spec`, `parameter_spec` or `compiled_foreman_spec`?

Reply:
Ah, when I put `entity_list` there (or `entity`, any *foreman*-specific type), `_flatten_entity` will bail out when trying to parse that.

Reply:
so I think my old logic was better ;)

Reply:
I've flipped back to the old `isinstance` type check and added a comment!